Sources/Moya/TargetType.swift

```diff
@@ -13,11 +13,8 @@ public protocol TargetType {
     /// The HTTP method used in the request.
     var method: Moya.Method { get }
 
-    /// The parameters to be encoded in the request.
-    var parameters: [String: Any]? { get }
-
-    /// The method used for parameter encoding.
-    var parameterEncoding: ParameterEncoding { get }
+    /// The default parameterEncoding for the `.encoded` RequestDataType case.
```

Comment:
I feel like we don't actually need to add this to the core. We do not use it anywhere, and this can be easily added to the migration guide as a best practice to use. Or better yet, make an extension to `ParameterEncoding`:
```swift
extension ParameterEncoding. {
    static var default: JSONEncoding {
        return JSONEncoding.default
    }
}
```

This way it would be easily used in task:
```swift
return .request(parameters: [:], encoding: .default)
```